### PR TITLE
Code generator record and enum updates.

### LINKED ISSUE #7167: override-method for records should suggest all overridable methods
### Description

When I want to override toString in a class, Netbeans supports that with a suggestion  to override toString, with all the
trimmings like the annotation, the method signature and visibility. The is nice.

For record that does not happen.

### Use case/motivation

I would like to get the suggestion to also override toString, equals, and hashCode for those cases where the default for
record does not cut it.

### Related issues

No 

### Are you willing to submit a pull request?

No

### PR DESCRIPTION
**Pull request by mbien**
Implement basic record code generation features.

image

records:

 - javac generates final methods for records unless they are already in the code
   - code can't override them which complicates things, but it can _replace_ them using the abstract method impl generator
 - made `equals`, `hashCode` and `toString` generators record aware
 - disabled getter/setter and property generator for records (record accessors are regular method overrrides)

enums:

 - `toString` generator supports now enums

fixes #7167

### CODE REVIEW

**Comment by mbien** [comment]
i went back and forth with implementing record accessors as getter generator or implements / override generator.

Ended up implementing it as override generator, since:
 - [JEP 395](https://openjdk.org/jeps/395) uses the term component accessors, getters are elements of the bean pattern and records are no beans
 - they are already there, so override makes most sense semantically (technically it is _replacing_ the impl since javac won't generate the synthetic method if it already exists)

I still don't like that the templates for simple tasks like those here are user customizable since there is nothing to customize other than removing a link.

**Review by matthiasblaesing — APPROVED** [review]
Looks good to me. Thank you!

File: java/java.editor/src/org/netbeans/modules/java/editor/codegen/ToStringGenerator.java [hunk 1]
```diff
@@ -303,4 +321,10 @@ private static MethodInvocationTree createAppendInvocation(TreeMaker make, Expre
         );
     }
 
+    private static ExpressionTree makeExpression(TreeMaker make, Element element) {
+        return element.getKind() == ElementKind.METHOD
+                ? make.MethodInvocation(Collections.emptyList(), make.MemberSelect(make.Identifier("this"), element), Collections.emptyList()) // NOI18N
+                : make.Identifier(element.getSimpleName());
+    }
```
Comment (mbien) [inline java/java.editor/src/org/netbeans/modules/java/editor/codegen/ToStringGenerator.java:328]: @lahodaj here I didn't know how to do a method invocation on implicit "this". So i cheated a bit after I saw that the doc says that `Identifier("this")` would do no validation.

Comment (lahodaj) [reply java/java.editor/src/org/netbeans/modules/java/editor/codegen/ToStringGenerator.java:328]: If you mean doing `componentName()`, then `make.MethodInvocation(Collections.emptyList(), make.Identifier(element.getSimpleName()), Collections.emptyList())` should work, I think.

**Review by lahodaj — APPROVED** [review]
Makes sense to me, thanks!

File: java/java.editor/src/org/netbeans/modules/java/editor/codegen/ToStringGenerator.java (hunk 1 above)
Comment (mbien) [reply java/java.editor/src/org/netbeans/modules/java/editor/codegen/ToStringGenerator.java:328]: oh wow that is easier. Don't know how I overlooked it. Working on an update if the thunderstorm allows it.

**Pull request merged**